### PR TITLE
Cv balldetection

### PR DESCRIPTION
**Pull request by Shrinandan-N**
This includes implementation of ball localization and angle calculations

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Comment (AG6GR) [inline .idea/.gitignore]: Can you add `.idea` to the top level `.gitignore` file? That should remove these temporary files.

File: ball_detection.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        return self.frame
+
+    def transform(self, x, y):
+        # transform coordinates to a system with the origin at the bottom-middle of screen
+        dims = self.frame.shape
+        x = x - (dims[1] / 2)
+        y = dims[0] - y
+        return x, y
+
+    def find_angle(self, x, y):
```
Comment (AG6GR) [inline ball_detection.py:48]: This is finding the angle with respect to the image right? But in reality we want the angle with respect to the robot. Might be helpful to draw a diagram from the side of the robot to visualize what these look like.

**Review by AG6GR — CHANGES_REQUESTED** [review]
Please put images in Drive, not in Github. As noted before, git is not well suited for handling large binary files such as images.

File: ball_detection.py (hunk 2 above)
Comment (AG6GR) [reply ball_detection.py:48]: We can talk more about this over the next week, doesn't have to be fixed in this PR.

File: ball_detection.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        cnts = imutils.grab_contours(cnts)
+
+        if len(cnts) > 0:
+            for cnt in cnts:
+                ((x, y), radius) = cv2.minEnclosingCircle(cnt)
+                if radius >= RADIUS_THRESH:
+                    # determine the num of border points on the contour (ex: differentiates between circle and rect)
+                    perimeter = cv2.arcLength(cnt, closed=True)
+                    borders = cv2.approxPolyDP(curve=cnt, epsilon=0.02 * perimeter, closed=True)
+                    if len(borders) > 6:
```
Comment (tsoim) [inline ball_detection.py]: magic num :(

File: run.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import cv2
+import numpy
+import ball_detection as bd
+
+img = cv2.imread("test_images/dis48_angle10.png")
```
Comment (tsoim) [inline run.py]: you could try and make this script take in an argument so you can pass in the filename. I like this package a lot: https://click.palletsprojects.com/en/8.0.x/ or there's also https://docs.python.org/3/library/argparse.html

**Review by AG6GR — APPROVED** [review]

**Pull request merged**